src/main/java/org/generationcp/middleware/service/DataImportServiceImpl.java

```diff
@@ -1189,4 +1197,30 @@ public int getMaxRowLimit() {
 	public void setMaxRowLimit(final int maxRowLimit) {
 		this.maxRowLimit = maxRowLimit;
 	}
+
+	private void validateTrialInstanceValueInObservation (final List<MeasurementRow> observations, final Map<String, List<Message>> errors) {
+		for (final MeasurementRow row : observations) {
+			for (final MeasurementData data : row.getDataList()) {
+				if (data.getMeasurementVariable().getTermId() == TermId.TRIAL_INSTANCE_FACTOR.getId()) {
+					final Optional<Message> message = Util.validateVariableValues(data.getMeasurementVariable(), data.getValue());
+					if (message.isPresent()) {
```

Comment:
if-else can be simplified as:
`errors.putIfAbsent(Constants.INVALID_TRIAL, new ArrayList<>());
					errors.get(Constants.INVALID_TRIAL).add(message.get());`